test/e2e/e2e_agent_test.go

```diff
@@ -274,6 +279,75 @@ func (p *plannerAgentLibvirt) RestartService() error {
 	return nil
 }
 
+func (p *plannerAgentLibvirt) Status() (*coreAgent.StatusReply, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	httpClient := &http.Client{
+		Transport: tr,
+	}
+
+	agentIP, _ := p.GetIp()
+	url := fmt.Sprintf("https://%s:3333/api/v1/status", agentIP)
+
+	res, err := httpClient.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("error getting agent status from local server: %v", err)
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response body: %v", err)
+	}
+
+	result := coreAgent.StatusReply{}
+	if err := json.Unmarshal(body, &result); err != nil {
+		return nil, fmt.Errorf("error decoding JSON: %v", err)
+	}
+
+	return &result, nil
+}
+
+func (p *plannerAgentLibvirt) Inventory() (*v1alpha1.Inventory, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	httpClient := &http.Client{
+		Transport: tr,
+	}
+
+	agentIP, _ := p.GetIp()
+	url := fmt.Sprintf("https://%s:3333/api/v1/inventory", agentIP)
+
+	res, err := httpClient.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("error getting agent inventory from local server: %v", err)
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response body: %w", err)
+	}
```

Comment:
This code is similar for functions `Version`, `Status` and `Login`, do you think we can have helper funtion for this, so we don't copy paste the code?

Reply:
Maybe even some kind of `sdk` for our API, with methods `Invetory`, `status`, `login`. Like:
```
agent := AgentApi(url)
agent.login()
agent.inventory()
agent.GetStatus()
```
or similar